gazebo/gui/SpaceNav.cc

```diff
@@ -147,8 +147,8 @@ void SpaceNav::Run()
   {
     msgs::Joystick joystickMsg;
 
-    // add button state
-    for (unsigned int i = 0; i < 2; ++i)
+    // add button state (26 bouttons)
+    for (unsigned int i = 0; i < 26; ++i)
       joystickMsg.add_buttons(this->dataPtr->buttons[i]);
```

Comment:
For users of the old spacenav version, this will populate 24 buttons that don't exist with zeroes, right? Is there a way we can avoid this?

Reply:
The best is detect the space nav version through the libspnav. But libspnav doesn't implement the possibility to get device id (Apparently in the daemon spacenav have all device id). 

Other possibilities
in SpaceNavPrivate.hh
we can declare  
std::vector<int> buttons;
int buttons[26];
and in SpaceNav.cc 
 case SPNAV_EVENT_BUTTON:
{
        // update button press
 if (sev.button.bnum >= buttons.size())
        {
            buttons.resize(sev.button.bnum +1);
         }
            this->dataPtr->buttons[sev.button.bnum] = sev.button.press;
            joystickMsg.mutable_buttons()->Set(sev.button.bnum, sev.button.press);
            this->dataPtr->joyPub->Publish(joystickMsg);
}